app/services/IdentityService.scala

```diff
@@ -43,35 +41,42 @@ class IdentityService(identityApiClient: IdentityApiClient) extends LazyLogging
     }
   }
 
-  def updateUserDetails(personalData: PersonalData, userId: UserId, authCookie: AuthCookie): Future[Unit] =
-    identityApiClient.updateUserDetails(
-      JsObject(Map("privateFields" -> personalData.convertToPrivateFields())), 
-      userId,
-      authCookie)
-    .map(_ => Unit)
+  def updateUserDetails(personalData: PersonalData, userId: UserId, authCookie: AuthCookie): Future[Unit] = {
+    val updatedFields =
+      createOnlyFields.foldLeft(personalData.convertToUser()) { (map, field) => map - field }
+
+    identityApiClient.updateUserDetails(updatedFields, userId, authCookie).map(_ => Unit)
+  }
 }
 
 object IdentityService extends IdentityService(IdentityApiClient) {
+  val primaryEmailAddress = "primaryEmailAddress"
+  val publicFields = "publicFields"
+  val createOnlyFields = Seq(primaryEmailAddress, publicFields)
```

Comment:
We can make them private, but they are actually used twice (in the definition of  `createOnlyFields` and in `convertToPrivateFields `)